### ISSUE TITLE
[SUGGESTION/FEATURE REQUEST] \Xgroupbystanza

### ISSUE THREAD

**Issue by lazzalazza** [maieul/ledmac#909]
# Description

It would be very nice (in some circumnstances even essential) to have a feature similar to \Xgroupbyline that worked for verses of the same stanza, grouping them in the same paragraph.

Such a command could work together with \Xgroupbyline (that could still be useful to specify that you don't want to have the same line number repeated).

I don't know if this is possible, or, for the time being, there is a workaround to get to a similar risult: in that case I might be very interested.

I don't think there's the need for a MWE, but I would gladly provide one if necessary. Once again, thanks for the great work and effort.

**Comment by maieul** [comment]
We always need a MWE. Always ! 

The point I don't see in your case, is in which way Xgroupbyline and
Xgroupbystanza could work together. Maybe an example of exepcted output
will help me.

Best

Maïeul

**Comment by lazzalazza** [comment]
I'm experiencing some trouble in providing you with an mwe, but for reasons that might be of interest to all: **the twocol mechanism very easily produces uneven columns**, such as that of the first file hereby attached. (I know that this is another issue, if you want I can open a new one).

```latex
\documentclass[11pt]{memoir}
\usepackage{fontspec} 

\usepackage[series={A},noend,noeledsec,noledgroup]{reledmac}
\fnpos{critical-familiar}
\Xarrangement[A]{twocol}
\Xhsizetwocol[A]{0.48\hsize}
\Xcolalign[A]{\justifying}
\Xlemmaseparator[A]{\textbf{:~}}
\AtEveryStopStanza{\vspace{.5\baselineskip}}
\setstanzaindents{0,0,0,0,0,0,0,0,0}
\numberstanzatrue
\Xgroupbyline                %%% To be changed hypothetically
                                       %%% in \Xgroupbystanza or something like that
\Xnumberonlyfirstinline
\Xstanza
\Xstanzaonlyfirst

\begin{document}
\beginnumbering
\stanza
Nel mezzo del \edtext{cammin}{%
  \Afootnote{first note}% 
} di \edtext{nostra}{%
  \Afootnote{other note on same line}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita\&

\stanza
Nel mezzo del \edtext{cammin}{%
  \Afootnote{first note}% 
} di \edtext{nostra}{%
  \Afootnote{other note on same line}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita&
Nel mezzo del cammin di \edtext{nostra}{%
  \Afootnote{note}% 
} vita\&
\endnumbering
\end{document}
```
In the first PDF I have the results of what reledmac produces now (!!!)
[mwe.pdf](https://github.com/maieul/ledmac/files/6215360/mwe.pdf)

In the second, I tried to achieve the result I was looking for by basically putting all the notes in the same Afootnote. Also here the two columns are uneven, and they should not be. I hope this at least works as a proof of concept.
[mwe (result).pdf](https://github.com/maieul/ledmac/files/6215382/mwe.result.pdf)

In the past message, I got confused between \Xgroupbyline and \Xnumberonlyfirstinline (it was late and I was kind of sleepy): it is \Xnumberonlyfirstinline (or \Xnumneronlyfirstintwolines) that should be able to work together with \Xgroupbystanza.

Thank you so much for your attention and sorry if I was not clear.

**Comment by maieul** [comment]
Indeed, there is a conflict between Xgroupbyline and column, at least with short note. I have some pist. See it later.

**Comment by lazzalazza** [comment]
Looking forward to it. In the meanwhile, thanks again for the amazing work: I (we) owe you a lot!

**Comment by maieul** [comment]
OKI, the bug is solved. For feature, I have done a preliminary work. 

But there is something complex in your suggestion.

With footnotes grouped by line, it is easy to know WHERE insert footnote in the flux of TeX footnote, and so decrease the remaning place in the page : at the end of each line.

But with footnotes grouped by stanza, should it be at the begining or at the end of the stanza ?

**Comment by lazzalazza** [comment]
I have downloaded the master version (was I supposed to do that?) but unfortunately it seems not to have solved the issue: sadly, even \Xstanzaonlyfirst doesn't work anymore. Here's the PDF output I get with lualatex.

[mwe.pdf](https://github.com/maieul/ledmac/files/6216527/mwe.pdf)

As for the footnotes grouped by stanza, I see the problem and I don't really know what's best. Ideally, it might work at the beginning: but it should probably be judged on the results.

Thank you so much: I'll have to really learn tex for good one day or another in the attempt of being more helpful.

**Comment by maieul** [comment]
It is not in master, but in issue909 branch (but don't try it now.)

For the problem... well, after will be (very) simplier to implement.
Before will be very complex, and requires more run...

**Comment by maieul** [comment]
And what about if a stanza in splitted on two pages ?

**Comment by maieul** [comment]
Ok, so the branch issue909 provides this feature (not for reledpar yet).  We insert it at the end of a stanza. But that create problem for stanza splitted in two pages... but at the beginining, it would also the same prroblem, and woul require more complex code and more runs. 

But, more I think to this idea, mor I think it is a bad idea... I think you should have instead one series by stanza,

**Comment by lazzalazza** [comment]
Sorry if I catch up now: I wasn't aware of the technical difficulties: of course, if the stanza is splitted in two pages, it would be bad to have notes that don't follow the text. I'm going to try the branch and see what happens! For now, thanks a lot!

**Comment by lazzalazza** [comment]
Dear Maïeul, I've tried the \Xgroupbystanza feature: let me tell you that is working AMAZINGLY good on the project I'm working on: I'm really impressed and grateful. PLEASE, DON'T REMOVE IT. I seriously think that you might consider keeping it in the package. Is there a way I can prove the usefullness of it?

**Comment by maieul** [comment]
Did you see the MWE with a stanza splitted on two pages ?

**Comment by lazzalazza** [comment]
Oh, now I see. The results were so impressive and eye-pleasing that I easily got carried away. I am sorry for not knowing enought tex to be able to help with the development directly (at least yet); therefore I don't know if making the package roughly "aware" of the position of verses in pages would be impossible, or so hard that even trying wouldn't have any sense.
On the other hand, reledmac is not only great for critical editions but also for writing commentaries, and such a feature would make it extremely convenient in that sense; using more than one series for doing that would unfortunately go against some typographical/philological traditions. Of course, time is a precious resource and I wouldn't want you to waste yours, so, this is your decision entirely. Just let me know if I could be of any help, maybe getting the chance to learn some hardcore tex in the meanwhile. ;) Thanks again.

**Comment by maieul** [comment]
Well, I have one théorical idea, but quite long to implement, and I am not sur it won't create problem of infinite loop.

So what I will do for now is 
- release bigfux
- keep open the issue, and maybe one day...
